components/match2/wikis/brawlhalla/match_legacy.lua

```diff
@@ -0,0 +1,102 @@
+---
+-- @Liquipedia
+-- wiki=brawlhalla
+-- page=Module:Match/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchLegacy = {}
+
+local Array = require('Module:Array')
+local Json = require('Module:Json')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper')
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+
+
+function MatchLegacy.storeMatch(match2, options)
+
+	if options.storeMatch1 then
+		local match = MatchLegacy._convertParameters(match2)
+
+		return mw.ext.LiquipediaDB.lpdb_match('legacymatch_' .. match2.match2id, Json.stringifySubTables(match))
+	end
+end
+
+function MatchLegacy._convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+
+	if match.walkover == 'FF' or match.walkover == 'DQ' then
+		match.resulttype = match.walkover:lower()
+		match.walkover = match.winner
+	elseif match.walkover == 'L' then
+		match.walkover = nil
+	end
+
+	match.staticid = match2.match2id
+	match.winner = nil
+
+	-- Handle extradata fields
+	local extradata = Json.parseIfString(match2.extradata)
+	match.extradata = {
+		matchsection = extradata.matchsection or '',
+	}
+
+	local bracketData = Json.parseIfString(match2.match2bracketdata)
+	if type(bracketData) == 'table' and bracketData.type == 'bracket' and bracketData.inheritedheader then
+		match.header = (DisplayHelper.expandHeader(bracketData.inheritedheader) or {})[1]
+	end
+
+	-- Handle Opponents
+	local handleOpponent = function(index)
+		local prefix = 'opponent' .. index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == Opponent.solo then
+			local player = opponentmatch2players[1] or {}
+			match[prefix] = player.name:gsub(' ', '_')
+			match[prefix .. 'score'] = (tonumber(opponent.score) or 0) > 0 and opponent.score or 0
+			match[prefix .. 'flag'] = player.flag
+			match.extradata[prefix .. 'displayname'] = player.displayname
+			match.extradata[prefix .. 'heads'] = player.char
+			if match2.winner == index then
+				match.winner = player.name
```

Comment:
possibly add some checks here to avoid the issue reported on discord:
`the player's name was too long and the truncation occured in the middle of one of the Japanese characters`

Reply:
Nah I'd just tell them to use v3 (match2) instead